components/org.wso2.carbon.secvault.securevault/src/main/java/org/wso2/carbon/secvault/securevault/SecureVaultInitializer.java

```diff
@@ -18,168 +18,130 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.wso2.carbon.secvault.securevault.config.model.SecureVaultConfiguration;
 import org.wso2.carbon.secvault.securevault.exception.SecureVaultException;
 import org.wso2.carbon.secvault.securevault.internal.SecureVaultConfigurationProvider;
 import org.wso2.carbon.secvault.securevault.internal.SecureVaultDataHolder;
 import org.wso2.carbon.secvault.securevault.internal.SecureVaultImpl;
-import org.wso2.carbon.secvault.securevault.reader.DefaultMasterKeyReader;
-import org.wso2.carbon.secvault.securevault.repository.DefaultSecretRepository;
+import org.wso2.carbon.secvault.securevault.model.SecureVaultConfiguration;
 
-import java.util.Optional;
-import java.util.ServiceLoader;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Path;
 
 /**
  * Responsible for initializing secure vault.
  *
- * @since 5.2.0
+ * @since 5.0.0
  */
 public class SecureVaultInitializer {
 
-    public boolean initialized = false;
-
     private static final Logger logger = LoggerFactory.getLogger(SecureVaultInitializer.class);
-    private static SecureVaultInitializer secureVaultInitializer = SecureVaultInitializer.getInstance();
-
-    private Optional<SecureVaultConfiguration> optSecureVaultConfiguration;
-
-    private String secretRepositoryType;
-    private String masterKeyReaderType;
-
-    private ServiceLoader<SecretRepository> secretRepositoryLoader;
-    private ServiceLoader<MasterKeyReader> masterKeyReaderLoader;
-
-    private Optional<String> secureVaultYAMLPath;
-    private Optional<String> masterKeyYAMLPath;
-    private Optional<String> secretPropertiesPath;
+    private static final SecureVaultInitializer SECURE_VAULT_INITIALIZER = new SecureVaultInitializer();
```

Comment:
Secure Vault Initializer will now keep a reference to the SecureVault instance. This is done because we are quite sure that there can be only a single SecureVault instance